Links redirect

While running `linkcheck`, I found several redirects for links and image sources. PR incoming to fix links, but not images, as a new image source should be used instead, preferably a locally hosted static image.

```
kitchen-sink/generic.rst:31: WARNING: redirect  http://www.python.org/ - permanently to https://www.python.org/
kitchen-sink/generic.rst:31: WARNING: redirect  http://www.python.org - permanently to https://www.python.org/
kitchen-sink/lists.rst:192: WARNING: redirect  http://www.yahoo.com - permanently to https://www.yahoo.com/
kitchen-sink/generic.rst:6: WARNING: redirect  https://source.unsplash.com/32x32/daily?icon - with Found to https://images.unsplash.com/photo-1528944442442-099aa4e85b27?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=32&ixid=MnwxfDB8MXxyYW5kb218MHx8aWNvbnx8fHx8fDE3MTQ5ODE0ODQ&ixlib=rb-4.0.3&q=80&utm_campaign=api-credit&utm_medium=referral&utm_source=unsplash_source&w=32
kitchen-sink/generic.rst:271: WARNING: redirect  https://source.unsplash.com/200x200/daily?cute+puppy - with Found to https://images.unsplash.com/photo-1437957146754-f6377debe171?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=200&ixid=MnwxfDB8MXxyYW5kb218MHx8Y3V0ZSBwdXBweXx8fHx8fDE3MTQ5ODE0ODQ&ixlib=rb-4.0.3&q=80&utm_campaign=api-credit&utm_medium=referral&utm_source=unsplash_source&w=200
kitchen-sink/admonitions.rst:20: WARNING: redirect  https://www.sphinx-doc.org/ - with Found to https://www.sphinx-doc.org/en/master/
(reference/kitchen-sink/blocks: line  171) broken    http://someurl/release-0.1.0.tar-gz - HTTPConnectionPool(host='someurl', port=80): Max retries exceeded with url: /release-0.1.0.tar-gz (Caused by NameResolutionError("<urllib3.connection.HTTPConnection object at 0x104c34bc0>: Failed to resolve 'someurl' ([Errno 8] nodename nor servname provided, or not known)"))
make: *** [linkcheckbroken] Error 1
```

--- a/sample-docs/kitchen-sink/admonitions.rst
+++ b/sample-docs/kitchen-sink/admonitions.rst
@@ -17,7 +17,7 @@ Sphinx provides several different types of admonitions.
    This is what admonitions are a special case of, according to the docutils
    documentation.
 
-   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/>`_, and more.
+   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/en/master/>`_, and more.
 
 ``admonition``
 ==============
@@ -26,7 +26,7 @@ Sphinx provides several different types of admonitions.
 
    It's got a certain charm to it.
 
-   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/>`_, and more.
+   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/en/master/>`_, and more.
 
 ``attention``
 =============
@@ -35,7 +35,7 @@ Sphinx provides several different types of admonitions.
 
    Climate change is real.
 
-   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/>`_, and more.
+   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/en/master/>`_, and more.
 
 ``caution``
 ===========
@@ -44,7 +44,7 @@ Sphinx provides several different types of admonitions.
 
    Cliff ahead: Don't drive off it.
 
-   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/>`_, and more.
+   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/en/master/>`_, and more.
 
 ``danger``
 ==========
@@ -53,7 +53,7 @@ Sphinx provides several different types of admonitions.
 
    Mad scientist at work!
 
-   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/>`_, and more.
+   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/en/master/>`_, and more.
 
 ``error``
 =========
@@ -62,7 +62,7 @@ Sphinx provides several different types of admonitions.
 
    Does not compute.
 
-   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/>`_, and more.
+   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/en/master/>`_, and more.
 
 ``hint``
 ========
@@ -71,7 +71,7 @@ Sphinx provides several different types of admonitions.
 
    Insulators insulate, until they are subject to ______ voltage.
 
-   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/>`_, and more.
+   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/en/master/>`_, and more.
 
 ``important``
 =============
@@ -80,7 +80,7 @@ Sphinx provides several different types of admonitions.
 
    Tech is not neutral, nor is it apolitical.
 
-   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/>`_, and more.
+   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/en/master/>`_, and more.
 
 ``note``
 ========
@@ -89,7 +89,7 @@ Sphinx provides several different types of admonitions.
 
    This is a note.
 
-   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/>`_, and more.
+   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/en/master/>`_, and more.
 
 ``seealso``
 ===========
@@ -98,7 +98,7 @@ Sphinx provides several different types of admonitions.
 
    Other relevant information.
 
-   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/>`_, and more.
+   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/en/master/>`_, and more.
 
 ``tip``
 =======
@@ -107,7 +107,7 @@ Sphinx provides several different types of admonitions.
 
    25% if the service is good.
 
-   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/>`_, and more.
+   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/en/master/>`_, and more.
 
 ``todo``
 ========
@@ -116,7 +116,7 @@ Sphinx provides several different types of admonitions.
 
    This needs the ``sphinx.ext.todo`` extension.
 
-   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/>`_, and more.
+   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/en/master/>`_, and more.
 
 ``warning``
 ===========
@@ -125,7 +125,7 @@ Sphinx provides several different types of admonitions.
 
    Reader discretion is strongly advised.
 
-   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/>`_, and more.
+   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/en/master/>`_, and more.
 
 ``versionadded``
 ================
@@ -134,7 +134,7 @@ Sphinx provides several different types of admonitions.
 
    Here's a version added message.
 
-   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/>`_, and more.
+   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/en/master/>`_, and more.
 
 ``versionchanged``
 ==================
@@ -143,7 +143,7 @@ Sphinx provides several different types of admonitions.
 
    Here's a version changed message.
 
-   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/>`_, and more.
+   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/en/master/>`_, and more.
 
 ``deprecated``
 ==============
@@ -152,4 +152,4 @@ Sphinx provides several different types of admonitions.
 
    Here's a deprecation message.
 
-   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/>`_, and more.
+   We also support *italic*, **bold**, ``code``, `links <https://www.sphinx-doc.org/en/master/>`_, and more.

--- a/sample-docs/kitchen-sink/generic.rst
+++ b/sample-docs/kitchen-sink/generic.rst
@@ -28,9 +28,9 @@ a :sup:`superscript`, :emphasis:`emphasis`, :strong:`strong`, and
 Hyperlinks
 ----------
 
-It is a website, so it'll have hyperlinks like http://www.python.org (inline),
+It is a website, so it'll have hyperlinks like https://www.python.org (inline),
 or Python_ (external reference), example_ (internal reference),
-`Python web site <http://www.python.org>`__ (external hyperlinks with embedded
+`Python web site <https://www.python.org>`__ (external hyperlinks with embedded
 URI), footnote references (manually numbered [1]_, anonymous auto-numbered [#]_,
 labeled auto-numbered [#label]_, or symbolic [*]_), citation references ([12]_),
 substitution references (|example|), and _`inline hyperlink targets`
@@ -199,7 +199,7 @@ Targets_.
 
 Explicit external targets are interpolated into references such as "Python_".
 
-.. _Python: http://www.python.org/
+.. _Python: https://www.python.org/
 
 Targets may be indirect and anonymous.  Thus `this phrase`__ may also
 refer to the Targets_ section.

--- a/sample-docs/kitchen-sink/generic.rst
+++ b/sample-docs/kitchen-sink/generic.rst
@@ -28,9 +28,9 @@ a :sup:`superscript`, :emphasis:`emphasis`, :strong:`strong`, and
 Hyperlinks
 ----------
 
-It is a website, so it'll have hyperlinks like https://www.python.org (inline),
+It is a website, so it'll have hyperlinks like https://www.python.org/ (inline),
 or Python_ (external reference), example_ (internal reference),
-`Python web site <https://www.python.org>`__ (external hyperlinks with embedded
+`Python web site <https://www.python.org/>`__ (external hyperlinks with embedded
 URI), footnote references (manually numbered [1]_, anonymous auto-numbered [#]_,
 labeled auto-numbered [#label]_, or symbolic [*]_), citation references ([12]_),
 substitution references (|example|), and _`inline hyperlink targets`

--- a/sample-docs/kitchen-sink/lists.rst
+++ b/sample-docs/kitchen-sink/lists.rst
@@ -189,13 +189,13 @@ Second list level
 ^^^^^^^^^^^^^^^^^
 
 - here is a list in a second-level section.
-- `yahoo <https://www.yahoo.com>`_
-- `yahoo <https://www.yahoo.com>`_
+- `yahoo <https://www.yahoo.com/>`_
+- `yahoo <https://www.yahoo.com/>`_
 
-  - `yahoo <https://www.yahoo.com>`_
+  - `yahoo <https://www.yahoo.com/>`_
   - here is an inner bullet ``oh``
 
-    - one more ``with an inline literally``. `yahoo <https://www.yahoo.com>`_
+    - one more ``with an inline literally``. `yahoo <https://www.yahoo.com/>`_
 
       heh heh. child. try to beat this embed:
 
@@ -204,8 +204,8 @@ Second list level
           :linenos:
           :lines: 10-20
 
-  - and another. `yahoo <https://www.yahoo.com>`_
-  - `yahoo <https://www.yahoo.com>`_
+  - and another. `yahoo <https://www.yahoo.com/>`_
+  - `yahoo <https://www.yahoo.com/>`_
   - ``hi``
 - how about an admonition?
 
@@ -217,15 +217,15 @@ Second list level
 But deeper down the rabbit hole
 """""""""""""""""""""""""""""""
 
-- I kept saying that, "deeper down the rabbit hole". `yahoo <https://www.yahoo.com>`_
+- I kept saying that, "deeper down the rabbit hole". `yahoo <https://www.yahoo.com/>`_
 
-  - I cackle at night `yahoo <https://www.yahoo.com>`_.
+  - I cackle at night `yahoo <https://www.yahoo.com/>`_.
 - I'm so lonely here in GZ ``guangzhou``
-- A man of python destiny, hopes and dreams. `yahoo <https://www.yahoo.com>`_
+- A man of python destiny, hopes and dreams. `yahoo <https://www.yahoo.com/>`_
 
-  - `yahoo <https://www.yahoo.com>`_
+  - `yahoo <https://www.yahoo.com/>`_
 
-    - `yahoo <https://www.yahoo.com>`_ ``hi``
+    - `yahoo <https://www.yahoo.com/>`_ ``hi``
     - ``destiny``
 
 Hlists

--- a/sample-docs/kitchen-sink/lists.rst
+++ b/sample-docs/kitchen-sink/lists.rst
@@ -189,13 +189,13 @@ Second list level
 ^^^^^^^^^^^^^^^^^
 
 - here is a list in a second-level section.
-- `yahoo <http://www.yahoo.com>`_
-- `yahoo <http://www.yahoo.com>`_
+- `yahoo <https://www.yahoo.com>`_
+- `yahoo <https://www.yahoo.com>`_
 
-  - `yahoo <http://www.yahoo.com>`_
+  - `yahoo <https://www.yahoo.com>`_
   - here is an inner bullet ``oh``
 
-    - one more ``with an inline literally``. `yahoo <http://www.yahoo.com>`_
+    - one more ``with an inline literally``. `yahoo <https://www.yahoo.com>`_
 
       heh heh. child. try to beat this embed:
 
@@ -204,8 +204,8 @@ Second list level
           :linenos:
           :lines: 10-20
 
-  - and another. `yahoo <http://www.yahoo.com>`_
-  - `yahoo <http://www.yahoo.com>`_
+  - and another. `yahoo <https://www.yahoo.com>`_
+  - `yahoo <https://www.yahoo.com>`_
   - ``hi``
 - how about an admonition?
 
@@ -217,15 +217,15 @@ Second list level
 But deeper down the rabbit hole
 """""""""""""""""""""""""""""""
 
-- I kept saying that, "deeper down the rabbit hole". `yahoo <http://www.yahoo.com>`_
+- I kept saying that, "deeper down the rabbit hole". `yahoo <https://www.yahoo.com>`_
 
-  - I cackle at night `yahoo <http://www.yahoo.com>`_.
+  - I cackle at night `yahoo <https://www.yahoo.com>`_.
 - I'm so lonely here in GZ ``guangzhou``
-- A man of python destiny, hopes and dreams. `yahoo <http://www.yahoo.com>`_
+- A man of python destiny, hopes and dreams. `yahoo <https://www.yahoo.com>`_
 
-  - `yahoo <http://www.yahoo.com>`_
+  - `yahoo <https://www.yahoo.com>`_
 
-    - `yahoo <http://www.yahoo.com>`_ ``hi``
+    - `yahoo <https://www.yahoo.com>`_ ``hi``
     - ``destiny``
 
 Hlists

